metrics/web/test/features/dashboard/presentation/widgets/text_metric_test.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:metrics/features/dashboard/presentation/widgets/text_metric.dart';
+
+void main() {
+  group("TextMetric", () {
+    testWidgets(
+      "can't be created with null title",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(const TextMetricTestbed(
+          title: null,
+          value: 'value',
+        ));
+
+        expect(
+          tester.takeException(),
+          isAssertionError,
+        );
+      },
+    );
+
+    testWidgets(
+      "can't be created with null value",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(const TextMetricTestbed(
+          value: null,
+          title: 'title',
+        ));
+
+        expect(
+          tester.takeException(),
+          isAssertionError,
+        );
+      },
+    );
+
+    testWidgets(
+      "displays the title text",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(const TextMetricTestbed());
+
+        expect(find.text(TextMetricTestbed.defaultTitleText), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "displays the value text",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(const TextMetricTestbed());
+
+        expect(find.text(TextMetricTestbed.defaultValueText), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "applies value and title text styles",
```

Comment:
What do you think about dividing this into two different test cases to follow SRP?